caput/tests/test_truncate.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
```

Comment:
yeah I guess it could be useful to test the behaviour of some edge cases, if only to be aware of their outcomes. These could include:
* precision set to 0/inf/NaN
* weight set to 0/inf/NaN
* value set to inf/NaN
* combinations of the above
* not an edge case exactly but truncating negative numbers

More generally a more robust test that I was using when developing these functions was to generate an ensemble of random numbers, truncate them, and check that the distribution of truncation errors was what you expect. That might be cumbersome and difficult to implement in this context though.

Another thing to note is there is a comment that claims one of the tests fails...

Reply:
Let's add a few special cases in for this.